### ISSUE TITLE
ENH: Python 3.13 Support for upstream Packages

### ISSUE THREAD

**Issue by CloseChoice** [shap/shap#3663]
## Problem Description

Python 3.13b1 is out, which means its time to start looking into Python 3.13 support!

Release wheels for python 3.13.

## Which dependencies support 3.13 ?

Core dependencies:
 - [x] numba: https://github.com/numba/numba/issues/9413
 - [x] pandas: https://github.com/pandas-dev/pandas/issues/58734
 - [x] numpy

Test dependencies:
 - [x] xgboost: https://github.com/dmlc/xgboost/pull/11028
 - [x] pytorch: should be supported, as of https://github.com/pytorch/pytorch/issues/130249
 - [x] lightgbm: https://github.com/microsoft/LightGBM/pull/6668
 - [x] catboost: https://github.com/catboost/catboost/issues/2748
 - [x] tensorflow: not supported (as of 2025-03-23), https://github.com/tensorflow/tensorflow/issues/78774
 - [x] ngboost: not supported (as of 2025-05-06), https://github.com/stanfordmlgroup/ngboost/blob/master/pyproject.toml#L16

**Comment by jameslamb** [comment]
> lightgbm: https://github.com/microsoft/LightGBM/pull/6668

To clarify.... `conda install -c conda-forge lightgbm` and `pip install lightgbm` will both totally work in Python 3.13 environments right now. That PR you've linked is just about adding more test coverage and a trove classifier.

For `shap`'s purposes, `lightgbm` already supports Python 3.13. The same is probably true for `xgboost` as well.

**Comment by CloseChoice** [comment]
@jameslamb you're right. I think we can start supporting python 3.13 as soon as #3861 passes.

**Comment by tr01-crypto** [comment]
any news?

**Comment by btat-qc** [comment]
@CloseChoice it looks like #3861 is passing according to your comment https://github.com/shap/shap/pull/3861#issuecomment-2854811847 Just wanted to ask what the status of 3.13 support is?

**Comment by CloseChoice** [comment]
@btat-qc yes, tests are passing and we support python 3.13 on master. I renamed this ticket to track the progress on Python 3.13 support on the remaining dependencies.

We can schedule a release quite soon. Would just like to fix a regression bug.

**Comment by CloseChoice** [comment]
All packages support python 3.13!
